src/effects/lv2/LV2Effect.h

```diff
@@ -339,16 +336,26 @@ class LV2Effect final : public LV2FeaturesList
 
 class LV2Wrapper final
 {
+   //! To compel use of the factory
+   struct CreateToken{};
 public:
    struct LV2Work {
       uint32_t size{};
       const void *data{};
    };
 
 public:
-   //! May spawn a thread
-   LV2Wrapper(const LV2FeaturesList &featuresList, int latencyPort,
+   //! Factory
+   static std::unique_ptr<LV2Wrapper> Create(
+      const LV2FeaturesList &featuresList,
+      const LV2Ports &ports, LV2PortStates &portStates,
+      const LV2EffectSettings &settings, float sampleRate, bool useOutput);
+
+   //! Constructor may spawn a thread
+   LV2Wrapper(CreateToken&&,
```

Comment:
I wonder if making the constructor private will do the trick as well

Reply:
Making the constructor private breaks make_shared or make_unique which have to invoke the constructor.

See guideline C.50 for an example of the trick: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-factory

Reply:
Makes sense!